### DISCUSSION TITLE
v1.37 -- Z2M pairing and connectivity performance problems - Deconz is fine, same devices

### DISCUSSION THREAD

**Discussion by Ltek** [Koenkk/zigbee2mqtt#22488] [General]
Encountering two major challenges in the 6+ months, after multiple versions of Z2M... and with 1.37 seems to be the worst so far with these.  I've recently added about 10 more devices that I moved from Deconz. I have multiple device makes/types and without exception they all pair with Deconz within 3-4 seconds but with Z2M some take multiple (3 to 5) attempts to pair. And I'm finding a few days after pairing, some simply fall off the map and don't communicate even when they show LQI of 240! See screenshot of one sensor literally 10ft through one wall away from SONOFF Zigbee 3.0 USB Dongle Plus-E (EZSP v12 7.3.1.0 build 176) ... LQI 240 but no connection -- so seem it is not communicating!?

![Z2M_device](https://github.com/Koenkk/zigbee2mqtt/assets/1626802/9dabb7e8-0014-474b-a2af-48c52b19f204)
![Z2M_map](https://github.com/Koenkk/zigbee2mqtt/assets/1626802/ca968837-48cc-4d71-98d6-b35c77663378)

**Comment by Outsidewall** [comment]
What zigbee director are you using?

**Reply by Ltek** [reply]
Coordinator is SONOFF Zigbee 3.0 USB Dongle Plus-E (EZSP v12 7.3.1.0 build 176)

I dont know what a director is, sorry - maybe you are asking something else?

**Reply by Outsidewall** [reply]
@Ltek, Sorry, meant coordinator.  I have had similar issues over the last year, and I have updated my dongle-e to the 7.4.2 firmware and configured z2m (1.37.0)  to use the (ember driver) , seems to be working well, better than it ever has.

**Reply by Ltek** [reply]
yesterday I updated to 7.4.3 (still on Z2M 1.37) and its "working kind of" a lot of device still have not reconnected after 18 hrs. Even though several are literally a few feet away from ones that are connected with LQIs over 200. Looks like I need to use 'Reconfigure' and wake them to get them to reconnect... I hope they fix that soon otherwise its back to Deconz. Z2M doesnt seem ready for prime time yet.

**Reply by francisp2** [reply]
> Z2M doesnt seem ready for prime time yet.

 you are using Dongle E as Coordinator - this device is specified as "experimental" and known as problematic

**Reply by Outsidewall** [reply]
Hello, I had to un-power a number of devices when I updated, for them to connect, once connected they have been good.

**Reply by Ltek** [reply]
I'm finding this to be the case... and almost always 'Reconfigure' them or they will not connect with the coordinator.

so far from what I'm seeing is not stable. Maybe its the hardware... not sure yet. 

But what I've been reading is that changing hardware is not easy with Z2m ... depending on the chipset you'll need to re-pair devices. This means Z2m is not a true (proper) abstraction layer between the hardware and the software. If it was, changing hardware to any chipset would not make any difference... the software holds the device configs, routes, etc.

**Reply by francisp2** [reply]
The dongles NVRAM holds the mesh network-addresses

**Reply by Outsidewall** [reply]
My research found that the reason people re-pair is that the ieee address of different dongles are different. However there are utilities that allow you to change the ieee address of the dongle. I have done this, I now have two dongle-e devices, with the same ieee address, only one is used at a time. This does allow me to test new firmware with out risking having to reset everything. You will also need to backup/restore the other settings in the dongle,  but again this is easily possible.

**Comment by Ltek** [comment]
> The dongles NVRAM holds the mesh network-addresses
What I read is Z2M holds the entire network, not the dongle.

The way to make this work (and IMO best way handle all/any hardware -- as has been done for decades with PCs and most hardware) is create a true abstraction layer. In Z2M's case, Z2M would own all network settings, including the dongle address that devices communicate with... 100% abstracting the dongle address so its irrelevant for backup/restore. The address the devices pair to is then 'virtual' and Z2M handles the dongle communications (in engineering its called 'address translation').

@Koenkk interested in your take.

**Reply by Koenkk** [reply]
That's not how coordinator firmwares work, I don't know why (maybe performance/latency reasons), but vendors implement a lot of functionality in their SDK, only a small portion goes over USB.

**Comment by Ltek** [comment]
> That's not how coordinator firmwares work, I don't know why (maybe performance/latency reasons), but vendors implement a lot of functionality in their SDK, only a small portion goes over USB.

I have both Z2M and Deconz (been using ~6 years)... deconz caches the device/network and changes the dongle ID automatically when you swap the dongle. So I suspect that hardware must be allowing the software to read / cache the device/route info, at least.   

For example, today I moved my Conbee 1 to new Conbee 3... took less than 10 minutes in totality.

1. Backup using web app
2. Shutdown Deconz
3. Swap USB dongle
4. Flash firmware on new dongle, just to be safe
5. Start Deconz  -- at this point (before even doing a network restore) I could see the entire network, it was all there so Deconz software must be caching all/most of it since the dongle was new.
6. Restore network using web app

**Reply by francisp2** [reply]
https://www.zigbee2mqtt.io/guide/faq/#requires-repairing

    Switching between adapters requires repairing, except when:
        When adapters are the same repairing is not required (e.g. CC2531 -> CC2531)
        Switching from a CC2530 or CC2531 based adapter to a CC2538/CC2652/CC1352 based adapter does not require repairing (e.g. CC2531 -> CC2652)
        Switching between CC2538, CC2652 and CC1352 based adapters does not require repairing (e.g. CC2538 -> CC2652)

**Comment by Ltek** [comment]
> https://www.zigbee2mqtt.io/guide/faq/#requires-repairing
> 
> ```
> Switching between adapters requires repairing, except when:
>     When adapters are the same repairing is not required (e.g. CC2531 -> CC2531)
>     Switching from a CC2530 or CC2531 based adapter to a CC2538/CC2652/CC1352 based adapter does not require repairing (e.g. CC2531 -> CC2652)
>     Switching between CC2538, CC2652 and CC1352 based adapters does not require repairing (e.g. CC2538 -> CC2652)
> ```
That's the exact reason I provided info that Conbee/Deconz combo doesn't require any device repair when changing Conbee dongles. They must be doing something that Z2M doesnt (I provided one hypothesis earlier). Which is why I pointed this out to @Koenkk 

There is clearly something that 'could be done' that Z2M is not doing so that changing adapters would be mostly seamless... aka, what Deconz is doing.

**Reply by francisp2** [reply]
> There is clearly something that 'could be done' that Z2M is not doing so that changing adapters would be mostly seamless... aka, what Deconz is doing.

In the time i have been using Zigbee2MQTT, I switched from a CC2531 to a CC2531 with antenna to a CC2530 to a CC2530+CC2591 to (for now) a CC2652, all without re-pairing anything.

**Comment by Ltek** [comment]
Maybe this will make it clear...
Conbee 1 is deRFmega256-23M12
Conbee 2 is ARM ATSAMR21B18
Conbee 3 is SiLabs EFR32MG21

... these are NOT the same manufacture or architecture and all can be migrated seamlessly from 1 to 2, 2 to 3, or 1 to 3 ... in seconds without network or device repair in Deconz.

**Comment by francisp2** [comment]
> these are NOT the same manufacture 

These are all made by Deconz, and Deconz is in total control of the firmware. Try using a Sonoff Dongle E (SiLabs EFR32MG21) with phoscon, won't work although it is the chipset as a Conbee III. Why ? Different firmware. And you can't flash Deconz firmware on it too, despite the same chipset.

**Reply by Ltek** [reply]
all Deconz code and FW flasher is open source. even if the FW is not open source, the way they are interacting with is it.  

If Conbee Deconz FW is different and more stable/reliable it would be best for Z2M community to embrace it. The Deconz devs might be open to sharing.

https://github.com/dresden-elektronik

https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/v2.27.0-beta
